sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3383,6 +3419,25 @@ object CleanupAliases extends Rule[LogicalPlan] {
     }
   }
 
+  // For struct field, it will be resolve as Alias(GetStructField, name),
+  // In Aggregate/GroupingSets this behavior will cause the same struct fields
+  // in aggExprs/groupExprs/selectedGroupByExprs be treated as different ones due to different
+  // ExprIds in Alias, and stops us finding the grouping expressions in aggExprs. Here we
+  // will use this method to remove Alias with different ExprId of GetStructField
+  // in groupByAlias/selectedGroupByExprs and aggregateExpressions
+  def trimStructFieldAlias(e: Expression): Expression = {
```

Comment:
It's only called in `ResolveReferences`, let's define it there.

Reply:
> It's only called in `ResolveReferences`, let's define it there.

Ok